task/tekton-operator-install/0.1/tekton-operator-install.yaml

```diff
@@ -0,0 +1,79 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: tekton-operator-install
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: pipelines, operator, tekton
+    tekton.dev/displayName: "Tekton Operator Install"
+spec:
+  description: >-
+    This task can be used to install Tekton pipelines and also it's 
+    components using Tekton Operator on a new cluster.
+  workspaces:
+    - name: kubeconfig
+      description: >-
+        The workspace consisting of kubeconfig file so that operator
+        can install tekton components on fresh cluster
+  params:
+    - name: TRIGGERS_VERSION
+      type: string
+      description: The version of the triggers to be installed
+      default: ""
+    - name: DASHBOARD_VERSION
+      type: string
+      description: The version of the Tekton dashboard to be installed
+      default: ""
+    - name: EXTENSION_WEBHOOKS_VERSION
+      type: string
+      description: >-
+        The version of the Tekton Extension Webhooks to be installed
+      default: ""
+  steps:
+    - name: intall-components
+      image: lachlanevenson/k8s-kubectl@sha256:3a5e22a406a109f4f26ec06b5f1f6a66ae0cd0e185bc28499eb7b7a3bbf1fe09
+      script: |
+        [[ -f $(workspaces.kubeconfig.path)/kubeconfig ]] && \
+        export KUBECONFIG=$(workspaces.kubeconfig.path)/kubeconfig || \
+        (echo "please provide kubeconfig file" && exit 0)
+
+        cat << EOF > trigger_cr.yaml
+        apiVersion: operator.tekton.dev/v1alpha1
+        kind: TektonAddon
+        metadata:
+          name: triggers
+        spec:
+          version: $(params.TRIGGERS_VERSION)
+        EOF
+
+        cat << EOF > dashboard_cr.yaml
+        apiVersion: operator.tekton.dev/v1alpha1
+        kind: TektonAddon
+        metadata:
+          name: dashboard
+        spec:
+          version: $(params.DASHBOARD_VERSION)
+        EOF
+
+        cat << EOF > extension_webhooks_cr.yaml
+        apiVersion: operator.tekton.dev/v1alpha1
+        kind: TektonAddon
+        metadata:
+          name: extensionwebhooks
+        spec:
+          version: $(params.EXTENSION_WEBHOOKS_VERSION)
+        EOF
+
+        kubectl apply -f https://storage.googleapis.com/tekton-releases/operator/latest/release.notags.yaml
+
+        if [[ -n $(params.TRIGGERS_VERSION) ]]; then
+          kubectl apply -f trigger_cr.yaml
+        elif [[ -n $(params.DASHBOARD_VERSION) ]]; then
+          kubectl create clusterrolebinding tekton-operator-cluster-admin \
+          --clusterrole cluster-admin --serviceaccount tekton-operator:tekton-operator
+          kubectl apply -f dashboard_cr.yaml
+        elif [[ -n $(params.EXTENSION_WEBHOOKS_VERSION) ]]; then
+          kubectl apply -f extension_webhooks_cr.yaml
+        fi
```

Comment:
I assume all 3 installation are independent so i feel we should not bind with if/elif condition :thinking: 

Even we can use switch instead of if condition

Reply:
But in switch we have two scenarios:-
1. `break` after executing the case means if we want to install two components then after installing one component it will break out from the switch thus not installing other component
2. `fallthrough` to next case which means that if suppose I want to install triggers then dashboard and other components also get's installed
WDYT?? :thinking:

Reply:
Should they all be separate `if`s?

Reply:
Yes if that's the case i agree with @JustinKuli we can use `if`s